src/NTDll.Desktop/NTDll.cs

```diff
@@ -23,5 +23,25 @@ public static partial class NTDll
         /// </remarks>
         [DllImport(nameof(NTDll))]
         public static extern Win32ErrorCode RtlNtStatusToDosError(NTSTATUS Status);
+
+        /// <summary>
+        /// The NtOpenSection routine opens a handle for an existing section object.
+        /// </summary>
+        /// <param name="sectionHandle">Pointer to a HANDLE variable that receives a handle to the section object.</param><param name="desiredAccess">Specifies an ACCESS_MASK value that determines the requested access to the object.</param><param name="objectAttributes">Pointer to an OBJECT_ATTRIBUTES structure that specifies the object name and other attributes. Use InitializeObjectAttributes to initialize this structure.</param>
+        /// <returns>
+        /// Returns <see cref="NTSTATUS.Code.STATUS_SUCCESS"/> on success, or the appropriate <see cref="NTSTATUS"/> error code on failure.
+        /// </returns>
+        [DllImport(nameof(NTDll), CharSet = CharSet.Unicode)]
+        public static extern NTSTATUS NtOpenSection([Out] out IntPtr sectionHandle, [In] uint desiredAccess, [In, Out] ref OBJECT_ATTRIBUTES objectAttributes);
```

Comment:
objectAttributes can only be a pointer if it's pinnable. It's not now because of its `string` field, but after my comments I think it will have to be a `char*` so it could then be a pointer here.
Making this parameter a pointer and using the `[Friendly]` attribute as @vbfox suggested will result in an overload being automatically generated that has no modifiers at all (no pointers, no `ref`) which will be more convenient for the user.